File: frontend/src/services/resources.js
import api from './api';

/**
 * Service for fetching external learning resources.
 */
const ResourceService = {
    /**
     * Gets 'Active Intel' (Analogy, Insight, Question) for a concept.
     * @param {string} conceptName 
     * @returns {Promise<Object>} { analogy: string, insight: string, question: string }
     */
    scoutResources: async (conceptName) => {
        return await api.get(`resources/scout/${encodeURIComponent(conceptName)}`);
    }
};

export default ResourceService;

File: src/routers/resources.py
from fastapi import APIRouter
from src.services.concept_tutor import ConceptTutorService

router = APIRouter(prefix="/api/resources", tags=["resources"])

@router.get("/scout/{concept_name}", summary="Get Active Intel for a concept")
async def scout_resources(concept_name: str):
    """
    Get 'Active Intel' (Analogy, Insight, Question) for a concept.
    Replaces old resource scout (books/courses).
    """
    return await ConceptTutorService.get_active_intel(concept_name)
